src/main/java/net/greatstart/controllers/UserController.java

```diff
@@ -58,14 +64,26 @@ public ModelAndView editProfile(@PathVariable long id, Principal principal) {
         return modelAndView;
     }
 
+    @PreAuthorize("isAuthenticated()")
     @PostMapping("/{id}/edit")
     public ModelAndView saveEditedProfile(@PathVariable long id,
                                           @Valid DtoUserProfile dtoUser,
                                           Errors errors,
-                                          Principal principal) {
+                                          Principal principal,
+                                          @RequestParam(value = "file", required = false) MultipartFile file) {
         if (errors.hasErrors()) {
             return new ModelAndView(EDIT_PROFILE);
         }
+        if (!file.isEmpty()) {
+            byte[] content = null;
+            try {
+                content = file.getBytes();
+                dtoUser.setPhoto(content);
+            } catch (IOException e) {
+                e.printStackTrace();
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Use a logger to log this exception. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS1148)